apps/web-giddh/src/app/ledger/components/newLedgerEntryPanel/newLedgerEntryPanel.component.ts

```diff
@@ -1487,14 +1497,27 @@ export class NewLedgerEntryPanelComponent implements OnInit, OnDestroy, OnChange
         if (this.currentTxn && this.currentTxn.selectedAccount && this.currentTxn.selectedAccount.accountApplicableDiscounts && this.currentTxn.selectedAccount.accountApplicableDiscounts.length) {
             this.currentTxn.selectedAccount.accountApplicableDiscounts.map(item => item.isActive = true);
             this.currentTxn.discounts.map(item => { item.isActive = false });
-            this.currentTxn.selectedAccount.accountApplicableDiscounts.forEach(element => {
-                this.currentTxn.discounts.map(item => {
-                    if (element.uniqueName === item.discountUniqueName) {
-                        item.isActive = true;
-                    }
-                    return item;
+            if(this.currentTxn.discounts && this.currentTxn.discounts.length === 1) {
```

Comment:
We are performing same logic in if else please check

Reply:
Yes but both will execute for diff condition
if there is newLedger entry panel discount not initi. then there will be 1 discount by default for value/percent in that case we need to update the discount array after discount componenet get init we will execute discount update array at last so implemented it with setTime so it get execute at last irrespective to component lifecycle